File: sieve.py
import sys
import math

def sieve(limit):
	primes = [True for i in range(limit)]
	for i in range(2, math.floor(math.sqrt(limit))):
		if (primes[i] and i * i < limit):
			for k in range(i*i, limit, i):
				primes[k] = False

	ret = 0
	for val in primes:
		ret += 1 if val else 0

	return ret - 2

if __name__ == "__main__":
	primeTo = int(sys.argv[1])
	ret = sieve(primeTo)
	print("Number of Primes from 0 to %d is %d" % (primeTo, ret))

File: sieve.c
#include <math.h>
#include <stdlib.h>

unsigned long sieve(long limit){
	unsigned long ret = 0;
	unsigned long computationalLimit = floor(sqrt(limit));
	unsigned char *prime = malloc(limit + 1);
	unsigned long i,j,k,l;

	for (i = 0; i < limit; i++){
		prime[i] = 1;
	}

	for (j = 2; j < computationalLimit; j++){
		if (prime[j] && j * j < limit){
			for (k = j * j; k <= limit; k += j){
				prime[k] = 0;
			}
		}
	}

	for (l = 0; l < limit; l++){
		ret += prime[l];
	}


	free(prime);
	//0, 1 are not primes
	return ret - 2;
}
